src/umpire/strategy/AllocationStrategy.hpp

```diff
@@ -120,12 +120,14 @@ class AllocationStrategy {
   friend std::ostream& operator<<(std::ostream& os,
                                   const AllocationStrategy& strategy);
 
+  virtual AllocationStrategy* getParentResource() const noexcept;
```

Comment:
I think we should call this `getParent` or `getParentAllocator`, since it might not be a resource.